Sources/FluentPostgreSQL/PostgreSQLDriver.swift

```diff
@@ -56,23 +58,7 @@ public class PostgreSQLDriver: Fluent.Driver {
 
         switch query.action {
         case .create:
-            // fetch the last inserted value
-            do {
-                let lastval = try _execute("SELECT LASTVAL();", [], connection)
-                // check if it contains an id
-                if let id = lastval[0, "lastval"]?.int {
-                    // return the id to fluent to
-                    // be the model's new id
-                    return .number(.int(id))
-                }
-            }
-            catch PostgreSQL.DatabaseError.invalidSQL(message: _) {
-                // When receiving a invalidSQL error, there is basically no LASTVAL, so ignore.
-            }
-
-            // no id found, return whatever
-            // the results are
-            return result
+            return (result[idKey]?.array as? [Node])?.first ?? result
```

Comment:
For some reason `RETURNING id` returns an array. I tried `LIMIT 1`, didn't make a difference. Maybe I am missing some SQL level trick which would allow returning a scalar value?